internal/fs/fs.go

```diff
@@ -1972,6 +1972,45 @@ func (fs *fileSystem) renameFile(
 	return nil
 }
 
+func (fs *fileSystem) handlePendingInodes(inodes *[]inode.DirInode) func() {
+	return func() {
+		for _, in := range *inodes {
+			fs.unlockAndDecrementLookupCount(in, 1)
+		}
+		*inodes = []inode.DirInode{}
+	}
+}
+
+func (fs *fileSystem) getBucketDirInode(ctx context.Context, parent inode.DirInode, name string, pendingInodes *[]inode.DirInode) (inode.BucketOwnedDirInode, error) {
+	dir, err := fs.lookUpOrCreateChildDirInode(ctx, parent, name)
+	if err != nil {
+		return nil, fmt.Errorf("lookup directory: %w", err)
+	}
+	*pendingInodes = append(*pendingInodes, dir)
```

Comment:
This is wrong. you are calling the method as getBucketDirInode and appending dir to pending inodes.

Reply:
Done

Reply:
Done